### PR TITLE
Add new zephyr workflows and documentation

### PR DESCRIPTION
**Pull request by MarkoSagadin**
## Description

This PR adds need Zephyr workflow files that can:
* Run the build during release process and capture the build artefacts,
* Run the build whenever a PR is opened, reopened or synchronised. 

The working example is already live in the https://github.com/IRNAS/irnas-zephyr-template repo.
The migration script from the previous PR can also migrate to this setup.

## Areas of interest for the reviewer

Check the added README, see if it is clearly written, if it can be reduced, etc.
Tell me if you are missing anything, I tried to think a bit more broadly 
regarding what can be automated, maybe I have missed some low hanging fruit.
Also check the `makefile` in the template repo, the main build logic is demonstrated there.

## After-review steps

- I will merge PR by myself.

[style guidelines]:
  https://github.com/IRNAS/irnas-guidelines-docs/blob/main/docs/developer_guidelines.md

### CODE REVIEW

**Comment by MarkoSagadin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-workflows-software/6" target="_blank">Graphite</a> 
    * **PR #7** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-workflows-software/7" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/IRNAS/irnas-workflows-software/7?utm_source=stack-comment).

File: workflow-templates/zephyr/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+Specifically, they provide:
+
+- Everything that [Basic](../basic/README.md) workflows already do: changelog
+  preparation, tagging, publishing GitHub releases, etc.,
+- Running _builds_ on Pull Requests and during releases processes,
+- Caching of West modules and toolchains downloaded by East to speed up the
+  _build_ process,
+- A way to change what _build_ means for each project,
+- A way to specify which artefacts are attached to the published GitHub releases
```
Comment (NejcKle) [inline workflow-templates/zephyr/README.md:17]: You are using artefacts and artifacts throughout this document, stick to one for consistency

File: workflow-templates/zephyr/README.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  creates a GitHub release with them.
+
+If anything goes wrong during `build.yaml` and `publish-release.yaml` workflows
+then the created release tag and Changelog update commit are deleted from the
+`main` branch.
+
+#### Pull requests
+
+`build.yaml` (and therefore _build_ process) is automatically triggered whenever
+a PR is opened, reopened or a new commit is pushed to it.
```
Comment (NejcKle) [inline workflow-templates/zephyr/README.md]: It triggers when a commit is pushed to the existing PR, right? Try wording this a bit more clearly.

File: workflow-templates/zephyr/README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+that is provided by `irnas-zephyr-template` repo by default.
+
+Note: all `make` commands are executed from the root of the cloned repository.
+Even if some command would `cd` into some other folder, the next command would
+still execute from the root.
+
+Expected behaviour of each `make` command:
+
+- `make install-dep` - Installs tooling needed by the project.
+- `make project-setup` - Set-ups the project and any tooling that might depend
```
Comment (NejcKle) [inline workflow-templates/zephyr/README.md]: Sets up

File: workflow-templates/zephyr/README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+- `make pre-package` - Only used in the release process, it is skipped if the
+  workflow was triggered due to a PR. Use it to package build artefacts.
+
+### Packaging build artefacts
+
+In release processes, the `build.yaml` will collect any files found in the
+`artefacts` folder of the project's root directory and attach them to the newly
+created a GitHub release as release artefacts.
+
+The developer can use `make pre-package` command to create the `artefacts`
```
Comment (NejcKle) [inline workflow-templates/zephyr/README.md]: can use the `make pre-package`...

**Review by NejcKle — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 5]
```diff
@@ -14,7 +14,6 @@ compile_commands.json
 .cache
 **/build/*
 build/*
-build*
```
Comment (TjazVracko) [inline .gitignore:17]: I would keep this.
On project with 2 MCUs I often have `build_nrf91` and `build_nrf52` folders, for example.

File: workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+name: Bug report
+about: Create a report on what is not working but it should
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
about: Create a report about something not working
```

File: workflow-templates/zephyr/.github/ISSUE_TEMPLATE/feature-request.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+---
+name: Feature request
+about: Normal development issue on technical matters
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/ISSUE_TEMPLATE/feature-request.md]:
```suggestion
about: Development issue describing something new to be added to the project
```

File: workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+title: BUG - <subject title related to the issue>
+labels: 'type: bug'
+assignees: ''
+
+---
+
+**Describe the bug**
+A clear and concise description of what the bug is.
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
<!--- A clear and concise description of what the bug is. --->
```

Maybe add comments like you have in the feature request?

File: workflow-templates/zephyr/.github/ISSUE_TEMPLATE/feature-request.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+<!--- Provide some context why is this feature needed.
+Does it build from an existing feature, does it enable an implementation of some other new feature? -->
+
+## Required steps / Implementation details
+
+<!--- Either provide a list of steps that are required for this feature and/or provide technical details on how this feature will be implemented. -->
+
+## Definition of Done
+
+<!--- Explicitly define conditions when this feature request is considered done, so it can be reviewed and validated. -->
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/ISSUE_TEMPLATE/feature-request.md:20]: Maybe add an example for this.

File: workflow-templates/zephyr/.github/pr-labeler.yml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+pull request: ['feature/*']
+release: ['release/*']
+fix: fix/*
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/pr-labeler.yml:4]: why is fix not in `[]`?

File: workflow-templates/zephyr/.github/pull_request_template.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+
+- [ ] My code follows the [style guidelines] as defined by BA-RD-AIS.
+- [ ] I have performed a self-review of my code.
+- [ ] My changes generate no new warnings.
+- [ ] I added/updated source code documentation for all newly added or changed
+      functions.
+- [ ] I updated all customer-facing technical documentation.
+
+Example strikeout: ~~- [ ] I updated all customer-facing technical
+documentation.~~ - This PR introduced only internal facing changes.
```
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/pull_request_template.md]:
```suggestion
Example strikeout: - [x] ~~I updated all customer-facing technical documentation.~~ - This PR introduced only internal facing changes.
```

This way the formatting of the list will not get borked.

File: workflow-templates/zephyr/.github/workflows/label_pr.yml
Comment (TjazVracko) [inline workflow-templates/zephyr/.github/workflows/label_pr.yml:1]: Which labels does this add? Maybe add a comment describing this

File: workflow-templates/zephyr/README.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+# Release notes
+
+{Contents of the pre_changelog.md}
+
+{Contents of the changelog for this version}
+
+{Contents of the post_changelog.md}
+```
+
+If some section is not used, its file can be empty. `pre_changelog.md` and
```
Comment (TjazVracko) [inline workflow-templates/zephyr/README.md]:
```suggestion
If a section is not used, the corresponding file can be empty. `pre_changelog.md` and
```

**Review by TjazVracko — COMMENTED** [review]
If i understand this correctly:

- `make build` calls `east release` for most of our firmware project?
- when I open or push to a PR, `east release` is run by the actions
- if `east release` fails, the PR gets that "red X can not merge" thing?

Otherwise, good stuff :+1:

File: .gitignore (hunk 5 above)
Comment (MarkoSagadin) [reply .gitignore:17]: This is just for this specific project, irnas-zephyr-template still has it.

File: workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md (hunk 8 above)
Comment (MarkoSagadin) [reply workflow-templates/zephyr/.github/ISSUE_TEMPLATE/bug_report.md]: Changed.

File: workflow-templates/zephyr/.github/pr-labeler.yml (hunk 10 above)
Comment (MarkoSagadin) [reply workflow-templates/zephyr/.github/pr-labeler.yml:4]: No specific reason, will make all the same.

**Comment by MarkoSagadin** [comment]
Thank you @TjazVracko and @NejcKle, I see that I made many grammatical mistakes :smiling_face_with_tear:
I fixed the requested changes, please have a look again.



> If i understand this correctly:
>
>    make build calls east release for most of our firmware project?

Yes, as dictated by the makefile.

>    when I open or push to a PR, east release is run by the actions

Yes, indirectly via makefile.

>    if east release fails, the PR gets that "red X can not merge" thing?

 Correct, but merging is not blocked, cause we don't have branch protection rules, due to free GitHub action.

> Otherwise, good stuff +1

Thank you, I hope this will be useful to us :smile:

**Review by NejcKle — APPROVED** [review]

**Review by TjazVracko — APPROVED** [review]
:+1:

**Pull request merged**